janitor/functions/case_when.py

```diff
@@ -99,26 +102,51 @@ def case_when(df: pd.DataFrame, *args, column_name: str) -> pd.DataFrame:
         `result` can be a scalar, a 1-D array, or a callable.
         If `result` is a callable, it should evaluate to a 1-D array.
         For a 1-D array, it should have the same length as the DataFrame.
-        The `default` argument applies if none of `condition0`,
-        `condition1`, ..., evaluates to `True`.
-        Value can be a scalar, a callable, or a 1-D array. if `default` is a
-        callable, it should evaluate to a 1-D array.
+    :param default: scalar, 1-D array or callable.
+        If callable, it should evaluate to a 1-D array.
         The 1-D array should be the same length as the DataFrame.
+        The element inserted in the output when all conditions
+        evaluate to False. Default is 0.
     :param column_name: Name of column to assign results to. A new column
         is created, if it does not already exist in the DataFrame.
-    :raises ValueError: If the condition fails to evaluate.
     :returns: A pandas DataFrame.
     """
-    conditions, targets, default = _case_when_checks(df, args, column_name)
+    check("column_name", column_name, [str])
+    default = if_else(df, args, default)
+    return df.assign(**{column_name: default})
+
+
+def if_else(df: pd.DataFrame, args, default: Optional[Any]) -> pd.Series:
```

Comment:
Whether renamed to `_if_else`, `if_else` isn't an accessor.

Reply:
this is so that it can be imported and used as a standalone within assign:

```py
from janitor.case_when import if_else
df.assign(a = if else(...), b = if else(...). ...)
```

Of course, if it is not a good idea, we can drop it

Reply:
dropping the `if_else` idea altogether; user can do multiple case_when if necessary, similar to what obtains in SQL. In the future, users can change this, if there is enough interest